xds/src/main/java/com/linecorp/centraldogma/xds/group/v1/XdsGroupService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.xds.group.v1;
+
+import static com.linecorp.centraldogma.server.internal.admin.auth.AuthUtil.currentAuthor;
+import static com.linecorp.centraldogma.server.internal.api.RepositoryServiceUtil.createRepository;
+import static com.linecorp.centraldogma.server.internal.api.RepositoryServiceUtil.removeRepository;
+import static com.linecorp.centraldogma.xds.internal.ControlPlanePlugin.XDS_CENTRAL_DOGMA_PROJECT;
+
+import com.google.protobuf.Empty;
+
+import com.linecorp.centraldogma.common.RepositoryExistsException;
+import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.metadata.MetadataService;
+import com.linecorp.centraldogma.server.storage.project.Project;
+import com.linecorp.centraldogma.server.storage.project.ProjectManager;
+import com.linecorp.centraldogma.xds.group.v1.XdsGroupServiceGrpc.XdsGroupServiceImplBase;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * An {@link XdsGroupServiceImplBase} implementation that provides methods to manage XDS groups.
+ */
+public final class XdsGroupService extends XdsGroupServiceImplBase {
+
+    private final ProjectManager projectManager;
+    private final CommandExecutor commandExecutor;
+    private final MetadataService mds;
+
+    /**
+     * Creates a new instance.
+     */
+    public XdsGroupService(ProjectManager projectManager, CommandExecutor commandExecutor) {
+        this.projectManager = projectManager;
+        this.commandExecutor = commandExecutor;
+        mds = new MetadataService(projectManager, commandExecutor);
+    }
+
+    @Override
+    public void createGroup(CreateGroupRequest request,
+                            StreamObserver<Group> responseObserver) {
```

Comment:
Optional) What do you think of using Reactive-gRPC instead of the vanilla gRPC? It is only used for server modules so we are relatively free of dependencies. `Reactive-gRPC` could remove boilerplate code such as `.onNext() + .onCopmlete()`.

Reply:
Thanks for the suggestion. 👍 
Well, honestly, I didn't want to use any other gRPC libraries because the logic is relatively simple (All methods are unary) and it might take more time (maybe just for me) when troubleshooting.

However, that doesn't mean that I will never use grpc-reactor. I will definitely use it when I think it gives more advantages or when I want to use it. 🤣